routing/index_graph_starter.cpp

```diff
@@ -76,10 +100,46 @@ void IndexGraphStarter::RedressRoute(vector<Joint::Id> const & route,
   }
 }
 
+void IndexGraphStarter::AddZeroLengthOnewayFeature(Joint::Id from, Joint::Id to)
+{
+  if (from == to)
+    return;
+
+  m_jointsOfAdditionalEdges.insert(from);
+  m_jointsOfAdditionalEdges.insert(to);
+  m_additionalZeroEdges.emplace_back(from, to, m_additionalNextFeatureId++);
+}
+
+void IndexGraphStarter::GetAdditionalEdgesList(Joint::Id jointId, bool isOutgoing,
+                                               vector<JointEdge> & edges)
+{
+  if (m_jointsOfAdditionalEdges.count(jointId) == 0)
+    return;
+
+  for (DirectedEdge const e : m_additionalZeroEdges)
+  {
+    if (isOutgoing)
+    {
+      if (e.GetFrom() == jointId)
+        edges.emplace_back(e.GetTo(), 0 /* weight */);
```

Comment:
I didn't quite understand the word "potential" in this context. Is your question if the A Star heuristic is overestimated or not? 

Here I connect two joints which have the same origin joint and correspond two points with exactly the same coords with a zero length edge. CalcHeuristic(e.GetTo(), e.GetFrom()) == 0. So the heuristic is admissible.

Start joint (or if the start is not a joint a joint closest to the start road point if go along to the edge) is connected with joints which originate (while adding restrictions process) from the start joint with zero length edge (if any). More or less the same happens with finish. This implemented in IndexGraphStarter::AddAdditionalZeroEdges(). 

Could you please explain which potential problem you see here?